internal/gcsx/file_cache_reader.go

```diff
@@ -133,14 +133,17 @@ func (fc *FileCacheReader) tryReadingFromFileCache(ctx context.Context, p []byte
 			switch {
 			case errors.Is(err, lru.ErrInvalidEntrySize):
 				logger.Warnf("tryReadingFromFileCache: while creating CacheHandle: %v", err)
+				err = nil
```

Comment:
Can we update all the variable, cacheHit, bytesRead as these can be used by defer statment?